cmd/handler-utils.go

```diff
@@ -27,7 +27,7 @@ import (
 // Validates location constraint in PutBucket request body.
 // The location value in the request body should match the
 // region configured at serverConfig, otherwise error is returned.
-func isValidLocationConstraint(r *http.Request) (s3Error APIErrorCode) {
+func isValidLocationConstraint(r *http.Request) (location string, s3Error APIErrorCode) {
```

Comment:
isValidLocationConstraint should never return `location`   change this function to take just the `location` string and parse the location outside in a different function as 

```go
func parseLocationConstraint(r *http.Request)  (location string, s3Error APIErrorCode) {
        // If the request has no body with content-length set to 0,
	// we do not have to validate location constraint. Bucket will
	// be created at default region.
	locationConstraint := createBucketLocationConfiguration{}
	err = xmlDecoder(r.Body, &locationConstraint, r.ContentLength)
        if err != nil && err != io.EOF {
 	       errorIf(err, "Unable to xml decode location constraint")
               return "", ErrMalformedXML
        }
        // Successfully decoded, proceed to verify the region.
	// Once region has been obtained we proceed to verify it.
	location = locationConstraint.Location
	if location == "" {
		// Location constraint is empty for region globalMinioDefaultRegion,
		// in accordance with protocol.
		location = globalMinioDefaultRegion
	}
        return location, nil 
}

func isValidLocation(location string) (s3Error APIErrorCode) {
         if serverConfig.GetRegion() != location {
                  return ErrInvalidRegion
         }
         return ErrNone
}

```